packages/fuel-gauge/src/funding-transaction.test.ts

```diff
@@ -507,12 +507,10 @@ describe('Funding Transactions', () => {
     await sleep(100);
 
     // Submitting TX 2 before TX 1 finished to process.
-    await expectToThrowFuelError(
-      () => fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId()),
-      new FuelError(
-        FuelError.CODES.INVALID_REQUEST,
-        'Transaction is not inserted. Hash is already known'
-      )
+    await expect(() =>
+      fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId())
+    ).rejects.toThrowError(
+      /Transaction input validation failed: Transaction id already exists \(id: .*\)/
```

Comment:
Shouldn't we validate if the error code is `FuelError.CODES.INVALID_REQUEST`?

Reply:
`expectToThrowFuelError` currently doesn't support this (from what I can tell).

Can implement this is required.

Reply:
Pseuco-code.

```ts
const { error } = await safeExec(async () => {
  fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId())
});

expect(error.code).toEqual(..)
expect(error.message).toEqual(..)
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3272/commits/5aabc578a2b5c5b5a30ab2ca3998b06a68138db0